ui/src/layouts/files.rs

```diff
@@ -141,27 +243,52 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                 div {
                     class: "files-list",
                     aria_label: "files-list",
-                    span {
-                        Folder {
-                            text: "Fake Folder 1".into(),
-                            aria_label: "fake-folder-1".into(),
-                        },
-                        File {
-                            text: "fake_2.png".into(),
-                            aria_label: "fake-file-1".into(),
-                        },
-                        Folder {
-                            text: "New Fake".into(),
-                            aria_label: "fake-folder-2".into(),
-                        },
+                    add_new_folder.then(|| {
+                        rsx!(
                         Folder {
-                            loading: true,
-                            text: "Fake Folder 1".into(),
-                        },
-                        File {
-                            loading: true,
-                            text: "Fake File".into(),
-                        }
+                            with_rename: true,
+                            onrename: |val| {
+                                ch.send(ChanCmd::AddNewFolder(val));
+                                ch.send(ChanCmd::GetItemsFromCurrentDirectory);
+                                add_new_folder.set(false);
+                             }
+                        })
+                    }),
+                    directories_list.read().iter().map(|dir| {
+                        rsx!(Folder {
+                            text: dir.name(),
+                            aria_label: dir.name(),
+                        })
+                    }),
+                    files_list.read().iter().map(|file| {
+                        rsx!(File {
+                            text: file.name(),
+                            aria_label: file.name(),
+                        })
+                    }),
+                    if STATIC_ARGS.use_mock {
```

Comment:
i'm fine with this but suspect that the mock data generation should instead go in the `testing/mock.rs` module